src/app/contextBroker/contextBroker.cpp

```diff
@@ -658,6 +660,15 @@ static void contextBrokerInit(void)
     QueueNotifier*  pQNotifier = new QueueNotifier(notificationQueueSize, notificationThreadNum, serviceV, serviceQueueSizeV, serviceNumThreadV);
     int             rc         = pQNotifier->start();
 
+    if (maxthreshold > 0)
+    {
+      maxthreshold=notificationQueueSize*maxthreshold/100;
```

Comment:
Note that notificationQueueSize refers to the size of the default queue size but you could have different per-service queues, each one with a diferent size (see `serviceQueueSizeV`).

Thus a global absolute max threshold cannot be established for all them.

Probably this code is not needed this. You can move this calculation of the the absolute max threshold to the part you have touched in src/lib/ngsiNotify/QueueNotifier.cpp

Reply:
> Note that notificationQueueSize refers to the size of the default queue size but you could have different per-service queues, each one with a diferent size (see `serviceQueueSizeV`).
> 
> Thus a global absolute max threshold cannot be established for all them.
> 
> Probably this code is not needed this. You can move this calculation of the the absolute max threshold to the part you have touched in src/lib/ngsiNotify/QueueNotifier.cpp

@fgalan, Thank you for your guidance. I will move this calculation code in ` src/lib/ngsiNotify/QueueNotifier.cpp` 
 I understood notificationQueueSize refers to the default queue size.
Could you please suggest which parameter should be used for getting the actual size of the queue.

Reply:
Updated in https://github.com/telefonicaid/fiware-orion/pull/4375/commits/1ebccb65531531a8dc0856d8de499b6a34d5a733